src/lib.rs

```diff
@@ -44,22 +43,33 @@ pub struct CausalSnapshot {
 
 impl CausalSnapshot {
     /// Construct a causal snapshot from a sequence of little endian bytes
-    pub fn from_le_bytes(words: [u32; 3]) -> Self {
-        let res_lsb = words[2] & core::u16::MAX as u32;
-        let res_msb = (words[2] >> 16) & core::u16::MAX as u32;
-        CausalSnapshot {
-            probe_id: words[0],
-            clock: words[1],
-            reserved_0: res_lsb as u16,
-            reserved_1: res_msb as u16,
+    pub fn from_le_bytes(words: [u32; 3]) -> Result<Self, ModalityProbeError> {
```

Comment:
There is only one error outcome possible in this function, so this error type seems way wide. The `InvalidProbeId`  type seems like a better fit.

Reply:
:+1: 91b6bb9122bdee6bd6578fd965b91f36190cda1f